omou-frontend/src/reducers/calenderReducer.js

```diff
@@ -0,0 +1,44 @@
+import initialState from './initialState';
+import * as actions from "../actions/actionTypes"
+
+export default function calender(state = initialState.CalenderData, { payload, type }) {
+    let newState = state;
+    let obj = newState.events_in_view;
+    switch (type) {
+        case actions.ADD_EVENT:
+            newState.events_in_view.push(payload);
+            console.log('CalenderData Action', payload, newState);
+            return state;
+
+        case actions.DELETE_EVENT:
+            for (let i = 0; i < newState.events_in_view.length; i++) {
+                let obj = newState.events_in_view[i];
+                if (payload.indexOf(obj.id) !== -1) {
+                    newState.events_in_view.splice(i, 1);
+                }
+            }
+            return state;
+
+        case actions.DELETE_ALL_EVENTS:
+            for (let i = 0; i < newState.events_in_view.length; i++) {
+                let obj = newState.events_in_view[i];
+                console.log(obj)
+            }
+            return state;
+
+        case actions.FILTER_EVENT:
```

Comment:
Users should also be able to filter events by course type (i.e. tutoring vs. course). As well as by room.

Reply:
To account for this logic, what you may want to do is make this function take 2 parameters (i.e. filter_key and filter_value). when you filter for specific objects, you can shorten this code to iterate through every object and keeping the ones where `eventInView[filter_key] === filter_value` . this would prevent you from having to have a million "if" statements.